app/controllers/application.js

```diff
@@ -52,17 +57,23 @@ export default Ember.Controller.extend({
     },
 
     willDestroy() {
-        Ember.$(document).off('keypress');
-        Ember.$(document).off('keydown');
+        if (!this.get('fastboot.isFastBoot')) {
+            Ember.$(window.document).off('keypress');
+            Ember.$(window.document).off('keydown');
+        }
     },
 
     stepFlash() {
-        this.set('flashError', this.get('nextFlashError'));
-        this.set('nextFlashError', null);
+        this.setProperties({
+            'flashError': this.get('nextFlashError'),
+            'nextFlashError': null
+        });
```

Comment:
Not related. I just grouped couple of individual set calls. Should've probably kept in a different commit.